### DISCUSSION TITLE
关于端口转发，不转发来源IP的逻辑问题

### DISCUSSION THREAD

**Discussion by yh317** [istoreos/istoreos#778] [Q&A] [answered]
关于端口转发的逻辑问题
（未找到类似issue）

### (1) 具体问题

端口转发，转发的数据包中不含数据包的来原公网IP，来源为路由器的IP。
这使得某些防火墙策略（如群晖通过来源IP屏蔽某些恶意攻击的IP地址）无效。
应该增加一个选项，转发数据包时是否保留来源IP地址。

### (2) 路由器型号和固件版本

E3845工控机
iStoreOS 21.02.3 2023042111 / LuCI istoreos-21.02 branch git-23.088.34636-f8cc111

### (3) 详细日志和/或截图

A：群晖后台登陆截图：
![Screenshot_2023-05-12-08-44-31-092_com microsoft emmx](https://github.com/istoreos/istoreos/assets/132770632/7c36a502-4b19-4260-900d-f850000b286a)

**Comment by jjm2473** [comment]
自己看看防火墙里面有没有这样的设置，如果没有，那就没有，我们不会去实现的。防火墙是openwrt的基础功能，如果你觉得不够用，应该去openwrt的项目提建议。

**Comment by jjm2473** [comment]
如果按你说的，那路由器自己也没办法访问群晖啊。

**Comment by yh317** [comment]
> 如果按你说的，那路由器自己也没办法访问群晖啊。

这个意思没明白。
就是端口转发应该保留来源IP。显示那几个10.0.0.1的IP实际上都是外网的IP，路由器转发的。

**Comment by jjm2473** [comment]
路由器转发以后，来源ip是不是变成路由器的内网IP了？如果这个路由器的内网IP都访问不了群晖，那不就代表路由器自己是访问不了群晖的吗？

**Comment by yh317** [comment, marked as answer]
是路由器转发外网数据包，但是来源IP并不是非得是路由器的。正常的情况应该能显示来源的公网IP地址。
我也找到解决方案了。在网络-防火墙里面配置一下nat规则就可以了。
在NAT规则里配置下“防止重写源”就可以了。
![Screenshot_2023-05-15-21-12-16-432_com microsoft emmx](https://github.com/istoreos/istoreos/assets/132770632/c5bf6479-be4d-426d-9d12-db248c2306f5)
配置好后，群晖的显示应该是这样的：
![Screenshot_2023-05-15-21-14-31-391_com microsoft emmx](https://github.com/istoreos/istoreos/assets/132770632/995e49cc-cff2-4cdf-94ad-3caf6d5c0e98)
虽然也是由路由器转发，但是可以识别到外网IP地址了。这样对某些IP的屏蔽规则就生效了。
总之，还是感谢大佬的帮助。
